Improve key mappings example

mickael-menu opened the pull request:
Clarify the key mappings example, including the basic LSP ones.

See https://github.com/mickael-menu/zk-nvim/discussions/17

mickael-menu commented on README.md:
````diff
@@ -355,13 +355,51 @@ require("zk.ui").get_pick_notes_list_api_selection(options)
+filetype plugin on
+```
+
+Then, create a new file under `~/.config/nvim/ftplugin/markdown.lua` to setup your Markdown key mappings.
````
This will enable the key bindings for all Markdown buffers, even if they are not in a notebook. Maybe there's a way to enable this only for `zk` notes? Adding a Lua API to determine if a path is part of a notebook?

kabouzeid replied:
`require("zk.util").notebook_root(path) ~= nil`

mickael-menu replied:
Thanks, it worked. I updated the README

kabouzeid commented:
Looks good overall. I have some thoughts, but have no time right now. I'll let you know in the evening.

mickael-menu commented:
@kabouzeid Do you want to address some stuff with this PR?

kabouzeid commented on README.md:
```diff
@@ -355,13 +355,52 @@ require("zk.ui").get_pick_notes_list_api_selection(options)
+  -- Preview a linked note.
+  map("n", "K", "<Cmd>lua vim.lsp.buf.hover()<CR>", opts)
+  -- Open the code actions for a visual selection.
+  map("v", "<leader>za", "<Cmd>'<,'>lua vim.lsp.buf.range_code_action()<CR>", opts)
```
This doesn't work. Must be `:'<,'>` instead of `<Cmd>'<,'>`.

kabouzeid commented on README.md:
```diff
@@ -355,13 +355,52 @@ require("zk.ui").get_pick_notes_list_api_selection(options)
+  -- Search for the notes matching a given query.
+  map("n", "<space>zf", "<Cmd>ZkNotes { match = vim.fn.input('Search: ') }<CR>", opts)
+  -- Search for the notes matching the current visual selection.
+  map("v", "<space>zf", ":'<'>ZkMatch<CR>", opts)
```
`:'<,'>`

kabouzeid replied:
why not `<leader>`?

kabouzeid commented on README.md:
```diff
@@ -355,13 +355,52 @@ require("zk.ui").get_pick_notes_list_api_selection(options)
+  -- Create a new note in the same directory as the current buffer, using the current selection for title.
+  map("v", "<leader>znt", ":'<'>ZkNewFromTitleSelection { dir = vim.fn.expand('%:p:h') }<CR>", opts)
+  -- Create a new note in the same directory as the current buffer, using the current selection for note content and asking for its title.
+  map("v", "<leader>znc", ":'<'>ZkNewFromContentSelection { dir = vim.fn.expand('%:p:h'), title = vim.fn.input('Title: ') }<CR>", opts)
```
`:'<,'>`

kabouzeid commented on README.md:
```diff
@@ -355,13 +355,52 @@ require("zk.ui").get_pick_notes_list_api_selection(options)
+  -- Create a new note in the same directory as the current buffer, after asking for its title.
+  map("n", "<leader>zn", "<Cmd>ZkNew { dir = vim.fn.expand('%:p:h'), title = vim.fn.input('Title: ') }<CR>", opts)
+  -- Create a new note in the same directory as the current buffer, using the current selection for title.
+  map("v", "<leader>znt", ":'<'>ZkNewFromTitleSelection { dir = vim.fn.expand('%:p:h') }<CR>", opts)
```
`:'<,'>`

kabouzeid commented:
Some mappings, like for example `ZkNotes`, `ZkNew`, `ZkTags`, `ZkMatch`, are also very useful outside of a notebook context.
In my experience, many (most?) users just copy and paste from the README.
It might not be the best experience to have them all only available when the buffer is a `zk` note.
Maybe first list the mappings, then below that, as a "pro tip", add that if one wants certain mappings to only be availabe in a `zk` note buffer, one could move them to a `ftplugin` and check for `require("zk.util").notebook_root`?


In the case of `ZkNew`, it wouldn't be a problem to overwrite with a more specific mapping when inside of a `zk` note buffer.
```lua
--- in main config
vim.api.nvim_set_keymap("n", "<leader>zn", "<Cmd>ZkNew { title = vim.fn.input('Title: ') }<CR>", opts)

--- ~/.config/nvim/ftplugin/markdown.lua
if require("zk.util").notebook_root(vim.fn.expand('%:p')) ~= nil then
  -- overwrite with a more specific mapping
  vim.api.nvim_buf_set_keymap(0, "n", "<leader>zn", "<Cmd>ZkNew { dir = vim.fn.expand('%:p:h'), title = vim.fn.input('Title: ') }<CR>", opts)
end
```

kabouzeid commented on README.md:
```diff
@@ -355,13 +355,52 @@ require("zk.ui").get_pick_notes_list_api_selection(options)
+  -- Open notes.
+  map("n", "<leader>zo", "<Cmd>ZkNotes<CR>", opts)
+  -- Open notes associated with the selected tags.
+  map("n", "<space>zt", "<Cmd>ZkTags<CR>", opts)
```
why not `<leader>`?

kabouzeid commented on README.md:
```diff
@@ -355,13 +355,52 @@ require("zk.ui").get_pick_notes_list_api_selection(options)
+  -- Open notes associated with the selected tags.
+  map("n", "<space>zt", "<Cmd>ZkTags<CR>", opts)
+  -- Open notes linking to the current buffer.
+  map("n", "<space>zb", "<Cmd>ZkBacklinks<CR>", opts)
```
why not `<leader>`?

kabouzeid commented on README.md:
```diff
@@ -355,13 +355,52 @@ require("zk.ui").get_pick_notes_list_api_selection(options)
+  -- Alternative for backlinks using pure LSP and showing the source context.
+  --map('n', '<leader>zb', '<Cmd>lua vim.lsp.buf.references()<CR>', opts)
+  -- Open notes linked by the current buffer.
+  map("n", "<space>zl", "<Cmd>ZkLinks<CR>", opts)
```
why not `<leader>`?

kabouzeid commented on README.md:
```diff
@@ -355,13 +355,52 @@ require("zk.ui").get_pick_notes_list_api_selection(options)
+  map("n", "<space>zl", "<Cmd>ZkLinks<CR>", opts)
+  
+  -- Search for the notes matching a given query.
+  map("n", "<space>zf", "<Cmd>ZkNotes { match = vim.fn.input('Search: ') }<CR>", opts)
```
why not `<leader>`?

kabouzeid commented on README.md:
```diff
@@ -355,13 +355,52 @@ require("zk.ui").get_pick_notes_list_api_selection(options)
+  -- Search for the notes matching a given query.
+  map("n", "<space>zf", "<Cmd>ZkNotes { match = vim.fn.input('Search: ') }<CR>", opts)
+  -- Search for the notes matching the current visual selection.
+  map("v", "<space>zf", ":'<'>ZkMatch<CR>", opts)
```
why not `<leader>`?

mickael-menu commented:
> Some mappings, like for example `ZkNotes`, `ZkNew`, `ZkTags`, `ZkMatch`, are also very useful outside of a notebook context.

You're perfectly right, I updated the README and fixed your other comments as well. Thanks!

kabouzeid commented:
lgtm :)
